lib/src/date-input/DateInput.tsx

```diff
@@ -46,10 +46,41 @@ const DxcDateInput = React.forwardRef<RefType, DateInputPropsType>(
     const [innerValue, setInnerValue] = useState(defaultValue);
     const [isOpen, setIsOpen] = useState(false);
     const [calendarId] = useState(`date-picker-${uuidv4()}`);
+    const [dayjsDate, setDayjsDate] = useState(getValueForPicker(value ?? defaultValue ?? "", format));
+    const [lastValidYear, setLastValidYear] = useState(
+      innerValue || value
+        ? !format.toUpperCase().includes("YYYY") && +getValueForPicker(value ?? innerValue, format).format("YY") < 68
+          ? 2000
+          : 1900
+        : undefined
+    );
     const colorsTheme = useTheme();
     const translatedLabels = useTranslatedLabels();
     const dateRef = useRef(null);
 
+    const getDate = (value) => {
+      if ((value || value === "") && format.toUpperCase().includes("YYYY")) return getValueForPicker(value, format);
+      else {
+        let newDate = getValueForPicker(value, format);
+        if (!lastValidYear) {
+          if (+newDate.format("YY") < 68) {
+            setLastValidYear(2000 + +newDate.format("YY"));
+            newDate = newDate.set("year", 2000 + +newDate.format("YY"));
+          } else {
+            setLastValidYear(1900 + +newDate.format("YY"));
+            newDate = newDate.set("year", 1900 + +newDate.format("YY"));
+          }
+        } else {
+          newDate = newDate.set("year", (lastValidYear <= 1999 ? 1900 : 2000) + +newDate.format("YY"));
+        }
+        return newDate;
+      }
+    };
+
+    useEffect(() => {
+      if (value || value === "") setDayjsDate(getDate(value));
+    }, [value]);
```

Comment:
`getDate` parameter is missing. If you want to avoid adding it, you can move outside the function `getDate` and pass the properties as references.